File: Assets/common/script/GenericScene.js
#pragma strict

class GenericScene extends MonoBehaviour{

	
	var skyboxCameraActive : boolean = false; //does this scene use a skybox camera?

	var networkObjects : Component[]; 
	var rpcOnly : boolean[];
	protected var separator : char[] = ["/"[0]];
	
	
	
	function Start () {
	
	}
	
	function Update () {
	
	}
	
	
	function MapSectorChanged(oldSector: Vector3, newSector : Vector3){}
	
	
	function ProcessOSCMessage(msg : OSCPacket){}
	
	function SendOSCMessage(){}
	
	/* send out osc messages for client screens */
	function configureClientScreens(){}
	
}

File: Assets/Standard Assets/Scripts/generic/GenericScene.cs
using UnityEngine;
using System.Collections;
using UnityOSC;


/* base class for scene related things */
public class GenericScene : MonoBehaviour {

		
	public bool skyboxCameraActive = false; //does this scene use a skybox camera?
	public bool sceneIsJumpInterruption = false;	//is this scene midway through a jump sequence? If so then turn this on. The jump plotting
													//display will then show display an override screen instead of text entry


	protected char[] separator = {'/'};



	public virtual void Start () {
		
	}

	public virtual void Update () {
		
	}

	/* called when the scene is changed */
	public virtual void LeaveScene(){}

	public virtual void MapSectorChanged(Vector3 oldSector , Vector3 newSector){}


	public virtual void ProcessOSCMessage(OSCPacket msg ){}

	public virtual void SendOSCMessage(){}

	/* send out osc messages for client screens */
	public virtual  void configureClientScreens(){}
		

}
